d4_ce/D4ConstraintEvaluator.cc

```diff
@@ -337,40 +325,39 @@ D4ConstraintEvaluator::slice_dimension(const std::string &id, const index &i)
 
 D4ConstraintEvaluator::index D4ConstraintEvaluator::make_index(const std::string &i)
 {
-    unsigned long long v = get_uint64(i.c_str());
-    return index(v, 1, v, false, false /*empty*/, "");
+    int64_t v = get_int64(i.c_str());
```

Comment:
I'm curious about why the change from uint64_t to int64_t It seems like in places we were misusing it?

Reply:
Yes, we were mixing signed and unsigned and given the issues we've found with buffer sizes, I thought it would be better to be conservative and consistent. At some point we, or someone else, might write a new libdap that can use uint64_t.